### PR TITLE
doc-gen: migrate scalar functions (string) documentation 1/4

### PR DESCRIPTION
**Pull request by Chen-Yuan-Lai**
## Which issue does this PR close?

Part of #13671 .

## What changes are included in this PR?

As discussed in #13671  , this PR will migrate the parts of builtin string functions documentation that currently support migration.

### CODE REVIEW

**Comment by Omega359** [comment]
I believe you may have forgotten to run the doc generation script prior to pushing this change

**Comment by Chen-Yuan-Lai** [comment]
@Omega359 Yes 😢 I'll run doc generation script for other PRs altogether. Thanks for reviewing 🙏

File: docs/source/user-guide/sql/scalar_functions.md [hunk 1]
```diff
@@ -852,10 +852,6 @@ btrim(str[, trim_str])
 
 #### Alternative Syntax
 
-```sql
-trim(BOTH trim_str FROM str)
-```
-
 ```sql
 trim(trim_str FROM str)
 ```
```
Comment (Chen-Yuan-Lai) [inline docs/source/user-guide/sql/scalar_functions.md:857]: @comphead  I found that `use_doc()` only supports one alternative syntax. Should we modify it to support multiple alternative syntaxes?

Comment (comphead) [reply docs/source/user-guide/sql/scalar_functions.md:857]: oh its a vec, yeah we should do it, as well as simplify a little bit the `Documentation` structure to replace `Option<Vec>` to just `Vec`

Comment (comphead) [reply docs/source/user-guide/sql/scalar_functions.md:857]: Lets leave trim not migrated for now and we can do it once the doc macros fixed

Comment (alamb) [reply docs/source/user-guide/sql/scalar_functions.md:857]: - I filed https://github.com/apache/datafusion/issues/13944 to track

Comment (Chen-Yuan-Lai) [reply docs/source/user-guide/sql/scalar_functions.md]: I remove the change for `trim()`  in this PR, thanks!  I'd like to update `use_doc()` and migrate `trim()` simultaneously

**Review by comphead — APPROVED** [review]
thanks @Chen-Yuan-Lai lgtm

**Pull request merged**